e2e/tests/negative/node_down.robot

```diff
@@ -0,0 +1,145 @@
+    And Check Longhorn node READY state on power off node is False
+    And Wait for statefulset 0 pod stuck in Terminating on the original node
+
+    When Power on off nodes
+    And Wait for longhorn ready
+    And Wait for statefulset 0 pods stable
+    And Check statefulset 0 data in file data is intact
+
+Power Off Node And Force Delete Terminating Statefulset Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}
+
+    When Create statefulset 0 using RWO volume with longhorn-test storageclass
+    Then Write 100 MB data to file data in statefulset 0
+    And Power off volume node of statefulset 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
+    And Wait for statefulset 0 pod is Running on another node
+    And Wait for statefulset 0 pods stable
+    And Check statefulset 0 data in file data is intact
+
+    When Power on off nodes
+    And Wait for longhorn ready
+
+Power Off Node And Not Force Delete Terminating Deployment Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}    numberOfReplicas=2
+
+    When Create persistentvolumeclaim 0 using RWO volume with longhorn-test storageclass
+    And Create deployment 0 with persistentvolumeclaim 0
+    Then Write 100 MB data to file data in deployment 0
+    And Power off volume node of deployment 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
+    And Wait for deployment 0 pod stuck in Terminating on the original node
+    And Wait for deployment 0 pod stuck In ContainerCreating on another node
+
+    When Force delete deployment 0 pod on the original node
+    And Wait for deployment 0 pod is Running on another node
+    And Wait for deployment 0 pods stable
+    And Check deployment 0 data in file data is intact
+
+    When Power on off nodes
+    And Wait for longhorn ready
+
+Power Off Node And Force Delete Terminating Deployment Pod
+    [Arguments]    ${node_down_pod_deletion_policy}
+    Given Set setting default-replica-count to 2
+    And Set setting node-down-pod-deletion-policy to ${node_down_pod_deletion_policy}
+    And Create storageclass longhorn-test with    dataEngine=${DATA_ENGINE}    numberOfReplicas=2
+
+    When Create persistentvolumeclaim 0 using RWO volume with longhorn-test storageclass
+    And Create deployment 0 with persistentvolumeclaim 0
+    Then Write 100 MB data to file data in deployment 0
+    And Power off volume node of deployment 0
+    And Sleep    360
+    And Check Longhorn node READY state on power off node is False
```

Comment:
1. Would using `Wait Until Keyword Succeeds` with a `15-second` interval be better?
```diff
     And Power off volume node of statefulset 0
-    And Sleep    360
-    And Check Longhorn node READY state on power off node is False
+    And Wait Until Keyword Succeeds    360s    15s    Check Longhorn node READY state on power off node is False
```

I think this approach would be more efficient as it can shorten the test case execution time. Moreover, this method is more dynamic and can detect state changes faster.

Reply:
Hi @roger-ryao , from the ([reference](https://longhorn.github.io/longhorn-tests/manual/pre-release/node/improve-node-failure-handling/)), script need wait approximately 6 minutes to wait pod change state to `Terminating` or `Running` on another node, in your suggestion, if `Check Longhorn node READY state on power off node is False` done before 360s(6 minutes), the remain time may not enough to wait for pod status change(Our default `RETRY_COUNT` is `300`), there will be four test case fail if using the suggestion https://ci.longhorn.io/job/private/job/longhorn-e2e-test/2471/

![image](https://github.com/user-attachments/assets/c4f0a42e-cba2-4432-bb02-06b754b4740b)